src/python/tritonfrontend/_api/_kservegrpc.py

```diff
@@ -90,47 +88,38 @@ def __post_init__(self):
             if isinstance(self.infer_compression_level, Grpc_compression_level):
                 self.infer_compression_level = self.infer_compression_level.value
 
+    @handle_triton_error
     def __init__(self, server: tritonserver, options: "KServeGrpc.Options" = None):
-        try:
-            server_ptr = server._ptr()  # TRITONSERVER_Server pointer
+        server_ptr = server._ptr()  # TRITONSERVER_Server pointer
 
-            # If no options provided, default options are selected
-            if options is None:
-                options = KServeGrpc.Options()
+        # If no options provided, default options are selected
+        if options is None:
+            options = KServeGrpc.Options()
 
-            if not isinstance(options, KServeGrpc.Options):
-                raise InvalidArgumentError(
-                    "Incorrect type for options. options argument must be of type KServeGrpc.Options"
-                )
+        if not isinstance(options, KServeGrpc.Options):
+            raise InvalidArgumentError(
+                "Incorrect type for options. options argument must be of type KServeGrpc.Options"
+            )
 
-            # Converts dataclass instance -> python dictionary -> unordered_map<string, std::variant<...>>
-            options_dict: dict[str, Union[int, bool, str]] = options.__dict__
+        # Converts dataclass instance -> python dictionary -> unordered_map<string, std::variant<...>>
+        options_dict: dict[str, Union[int, bool, str]] = options.__dict__
 
-            self.triton_frontend = TritonFrontendGrpc(server_ptr, options_dict)
-        except TritonError:
-            exc_type, exc_value, _ = sys.exc_info()
-            # raise ... from None masks the tritonfrontend Error from being added in traceback
-            raise ERROR_MAPPING[exc_type](exc_value) from None
+        self.triton_frontend = TritonFrontendGrpc(server_ptr, options_dict)
 
     def __enter__(self):
         self.triton_frontend.start()
         return self
 
+    @handle_triton_error
     def __exit__(self, exc_type, exc_value, traceback):
         self.triton_frontend.stop()
         if exc_type:
-            raise ERROR_MAPPING[exc_type](exc_value) from None
+            raise exc_type(exc_value) from None
```

Comment:
Same comment: https://github.com/triton-inference-server/server/pull/7720#discussion_r1811533149

Why did this one keep the `from None` but the others didn't?